pandas/tests/io/pytables/test_store.py

```diff
@@ -385,6 +385,10 @@ def test_to_hdf_with_min_itemsize(tmp_path, setup_path):
     tm.assert_series_equal(read_hdf(path, "ss4"), concat([df["B"], df2["B"]]))
 
 
+@pytest.mark.xfail(
+    using_string_dtype() and has_pyarrow,
+    reason="TODO(infer_string): can't encode '\ud800': surrogates not allowed",
+)
```

Comment:
Do you know if the error happens when writing or reading? (I suppose reading because even if it was object dtype we try to read it as strings?)

This is probably not something we can fix? Wondering (not for this PR) if we should fall back to object dtype in such a case (or at least give that option), so the user can still load the data and inspect it.

Reply:
I hadn't looked into this one - currently it raises on data creation. Fixing that by specifying object dtype, you're right it's the reading that raises here. The error happens in `Series.str.decode(encoding, errors=errors)`. I am wondering if we should add a `dtype` argument to `decode` so that you can force object dtype - that seems generally useful to alleviate this pain-point, and then we could use it to provide a fallback in pytables.

Reply:
PyArrow strings not supporting surrogates is another interesting case in regards to PDEP-13. #58455

Reply:
> I am wondering if we should add a `dtype` argument to `decode` so that you can force object dtype - that seems generally useful to alleviate this pain-point, and then we could use it to provide a fallback in pytables.

+1

Reply:
I listed this issue in https://github.com/pandas-dev/pandas/issues/59328 as one of the breaking changes when users upgrade to the string dtype (in this case at least when using pyarrow). 

But so we should also list this as one of the differences between the pyarrow and python engines.

Reply:
I do wonder if we should start nudging users down the path of referring to our strings as utf8. For other encodings, the binary type is the best bet for now, and probably for a while (I'm not aware of any non-utf8 Arrow array work, but maybe its out there)